App: Add MCUBoot update support (#265)
nrfconnect/ncs-serial-modem: doc/app/at_fota.rst

* The ``<file url>`` parameter is a string.
  It represents the full HTTP or HTTPS path of the target image to download.
  It must be provided to the start operations.
  For the MCUboot bootloader update, the target image must be the signed image for the **inactive** ``s0``/``s1`` slot.

>   ``AT#XBOOTINFO=1`` can be used to query which slot is active.

Reviewer: ```suggestion
  You can use the ``AT#XBOOTINFO=1`` command to query which slot is active.
```

   Activating the new modem firmware is done identically to a modem delta update, by resetting either the whole device or only the modem.

.. note::

   For the MCUboot bootloader update, NSIB (B0) uses monotonic counter values to prevent rollback.

>    This means that MCUboot bootloader updates are limited to the number of ``SB_CONFIG_SECURE_BOOT_NUM_VER_COUNTER_SLOTS`` slots. In |SM| this is 40 by default.

Reviewer: ```suggestion
      This means that updates for MCUboot images are limited to the number of slots defined in the ``SB_CONFIG_SECURE_BOOT_NUM_VER_COUNTER_SLOTS`` Kconfig option. 
   In |SM|, it is set to ``40`` by default.
```
